mlc_llm/core.py

```diff
@@ -366,9 +375,14 @@ def _parse_args(parsed) -> argparse.Namespace:
             "tvm.contrib.vllm.single_query_cached_kv_attention", True
         ), "TVM needs to be built with -DUSE_VLLM=ON."
 
-    parsed.artifact_path = os.path.join(
-        parsed.artifact_path, f"{parsed.model}-{parsed.quantization.name}"
-    )
+    model_name = [
+        parsed.model,
+        parsed.quantization.name,
+    ]
+    if parsed.use_presharded_weights:
+        model_name.append(f'presharded-{parsed.num_shards}gpu')
+
+    parsed.artifact_path = os.path.join(parsed.artifact_path, '-'.join(model_name))
```

Comment:
Thank you, and updated.  Since `mlc-llm` currently doesn't have a CI lint step, I've been disabling my local use of `black` to avoid causing lots of re-formating churn.  Unfortunately, that means I've also been contributing to the problem of having unformatted code, and I don't yet have a solution to only format new code.